### ISSUE TITLE
Don't generate SwaggerException

### ISSUE THREAD

**Issue by JanneHarju** [RSuter/NSwag#2001]
It would be nice that ether there is no export for SwaggerException or parameter how can I set one type excluded from export. This is problem when I'm generating one Angular service per one controller. When I use these services I need to modify generated code by removing export manually from SwaggerException. Problem is that every service has its own SwaggerException. So I cannot use index.ts for exporting services from module. If I set SwaggerException in excludedTypeNames code get broken because there is still parts where SwaggerException is used.

So ether I need way to exclude some types from export list. It could be combination of excludedTypeNames and exportTypes. So list where I can specify which Types I don't want to export.

Other solution is that there is no exception handling in service in first place. (or parameter for this)

And last solution is that you use HttpErrorResponse from @angular/common/http instead of SwaggerException.

**Comment by RSuter** [comment]
You can exclude it with the ExcludedTypeNames setting

**Comment by JanneHarju** [comment]
As I said.
"If I set SwaggerException in excludedTypeNames code get broken because there is still parts where SwaggerException is used."

So it will exlide SwaggerException

**Comment by JanneHarju** [comment]
Sorry I acidentally closed this.

**Comment by RSuter** [comment]
This exception is always used by the generated code... with the exclude option you can only disable the generation of the class itself... is the problem that it is trying to export it? If the import of your custom implementation is missing then you need to create an extension code file and add the custom import there

**Comment by JanneHarju** [comment]
Yes my problem is that SwaggerException is exported always.

One option could be that I can control what classes will not be exported. So it works like exportTypes boolean, but instead of this it would be excludeExportTypes. In that wat SwaggerException will be there but it won't be exported. Configuration could be this.

`excludeExportTypes: [ "SwaggerException" ]`

Problem is that it is much easier that I can exclude exports than list all exported classess and stuff from service. My Angular library's public_api.ts is looking this.

```
export * from './lib/services/Configuration-service';
export * from './lib/services/Localization-service';
```

So if both services will export SwaggerException here will be conflict.

I didn't get this:

> If the import of your custom implementation is missing then you need to create an extension code file and add the custom import there

**Comment by crashsol** [comment]
i don't want generate the status code check too ,because the fetch  features do not throw exception when status code like 400,401 。。。i want do the logic in my code #2006

**Comment by Jestar342** [comment]
> This exception is always used by the generated code...

That's what is being challenged: Why? The very core fundamental of HTTP is the response codes. Why does NSwag assume an exception for everything that is >399, or sometimes worse.. not 200 or 204?

Anyway.. that's what is being asked for here. This is a feature request, not a question. Ability to _disable_ exception generation. We can already wrap responses, and it seems fundamentally broken that this only applies to what NSwag arbitrarily considers a "success".

**Comment by RicoSuter** [comment]
> Why does NSwag assume an exception for everything that is >399, or sometimes worse.. not 200 or 204?

Mainly because C# can only have one return type and we tried to generate idiomatic C# - that's why we assume that everything except 2xx is an exception. This is how the templates work, i.e. we try to abstract away HTTP so that we only have "plain" C# interface - if you don't like you should look at AutoRest which generates more HTTP specific code I think.

**Comment by Jestar342** [comment]
try
            {
                var result = await client.GenerateAsync(parameters.Foo, parameters.Bar);

                return result.StatusCode == (int) HttpStatusCode.OK
                    ? result.Result
                    : null;
            }
            catch (ApiException)
            {
                return null;
            }

That `try {} catch {}` is what is clunky when wrapping the responses. Infact I argue it makes the entire feature of wrapping of responses pointless.

**Comment by RicoSuter** [comment]
Yes, for wrapping of responses it would make sense to not throw exceptions..
